uxarray/grid.py

```diff
@@ -160,33 +160,57 @@ def __from_ds__(self, dataset):
             raise RuntimeError("unknown file format: " + self.mesh_filetype)
         dataset.close()
 
-    def write(self, outfile, grid_type):
+    def write(self, outfile, grid_type, save_as='netcdf'):
         """Writes mesh file as per extension supplied in the outfile string.
 
         Parameters
         ----------
         outfile : str, required
-            Path to output file
+            Name of or path to output file. If path is not given,
+            file is saved to current working directory
+
+            ex: "path/to/new/file/location/new_file.nc" or
+                "new_file.nc"
+
         grid_type : str, required
             Grid type of output file.
             Currently supported options are "ugrid", "exodus", and "scrip"
 
+        save_as : str, default "netcdf"
+            The specific file type to save newly created datasets to.
+            Current options are "netcdf", "zarr", and "None" (no file saved)
+
         Raises
         ------
         RuntimeError
-            If unsupported grid type provided or directory not found
+            If unsupported provided grid type, file type, or outfile directory is not found
         """
 
         if grid_type == "ugrid":
-            _write_ugrid(self.ds, outfile, self.ds_var_names)
+            out_ds = _encode_ugrid(self.ds)
+
         elif grid_type == "exodus":
-            _write_exodus(self.ds, outfile, self.ds_var_names)
+            out_ds = _encode_exodus(self.ds, outfile, self.ds_var_names)
+
         elif grid_type == "scrip":
-            _write_scrip(outfile, self.Mesh2_face_nodes, self.Mesh2_node_x,
-                         self.Mesh2_node_y, self.face_areas)
+            out_ds = _encode_scrip(self.Mesh2_face_nodes, self.Mesh2_node_x,
+                                   self.Mesh2_node_y, self.face_areas)
         else:
             raise RuntimeError("Format not supported for writing: ", grid_type)
 
+        if save_as == 'netcdf':
+            out_ds.to_netcdf(outfile)
```

Comment:
I think we need to find a way to support the whole signature of Xarray.Dataset.to_netcdf (`Dataset.to_netcdf(path=None, mode='w', format=None, group=None, engine=None, encoding=None, unlimited_dims=None, compute=True, invalid_netcdf=False)`)

Reply:
IMO the user could easily do `grid.encode_as("scrip").to_netcdf(...)` where `grid.encode` returns a Dataset. That means uxarray doesn't have to deal with all the details and file format backends that keep changing

Reply:
I agree. @michaelavs please let me know if you can do this tomorrow; otherwise, I can do it.

Reply:
I have made it this way